src/reactapp/src/components/CheckoutForm.jsx

```diff
@@ -15,20 +15,37 @@ import StickyRightSidebar from './StickyRightSidebar';
 import CheckoutAgreements from './checkoutAgreements';
 import CheckoutFormWrapper from './CheckoutFormWrapper';
 import { config } from '../config';
+import { aggregatedQueryRequest } from '../api';
 import useAppContext from '../hook/useAppContext';
 import useCartContext from '../hook/useCartContext';
 
 function CheckoutForm() {
-  const { orderId, getGuestCartInfo } = useCartContext();
-  const [{ pageLoader }, { setPageLoader }] = useAppContext();
+  const { orderId, storeAggregatedCartStates } = useCartContext();
+  const [{ pageLoader }, appActions] = useAppContext();
+  const {
+    setPageLoader,
+    dispatch: appDispatch,
+    storeAggregatedAppStates,
+  } = appActions;
 
   useEffect(() => {
     (async () => {
-      setPageLoader(true);
-      await getGuestCartInfo();
-      setPageLoader(false);
+      try {
+        setPageLoader(true);
+        const data = await aggregatedQueryRequest(appDispatch);
+        storeAggregatedCartStates(data);
+        storeAggregatedAppStates(data);
+        setPageLoader(false);
```

Comment:
I dont think the `dispatch` function comes from the react reducer really allows us to control its operation synchronously. I just put `await` for the two-state update calls. But these operations happen very fast and `setPageLoader` is also makes a state change and we do not see much difference even with the `await`

Reply:
Sure perhaps now and in most cases/systems this code runs very fast and a difference wont be noticed. But we cant predict how the code will evolve in the future or which systems are the target, they may be very low performance. Right now it is not robust enough to support these edge cases without potential for error.

Putting await in front of the `storeAggregatedCartStates` and `storeAggregatedAppStates` functions is one way to solve this, but then you lose the benefits of asynchronously handling them. Instead I am suggesting you create an array of promises in which you store these function calls and call `Promise.all()` to guarantee both functions are handled (asynchronously) before `setPageLoader(false)`.

ex:
```js
const promises = [];
promises.push(storeAggregatedCartStates(data));
promises.push(storeAggregatedAppStates(data));
await Promise.all(promises);
setPageLoader(false);
```

Reply:
Still, I feel this won't make any difference. because `dispatch` cannot be asynchronously managed as far as i know. We cannot do synchronous state update with react hooks. This was possible with setState where we can pass a callback function as a second param which will be executed after the states specified in the first params are updated.

Basically the `await` call and the `Promise. all` would not have any effect. 

If we can control this somehow, then I like to keep them synchronously using `await`. because it is very important to update the cart states first before app state. Otherwise, we will face unwanted issues in the app. You will see errors if you change the order of those state updates. ie try to update app states first then cart states.

Reply:
Ok we can keep it as is